text_generation/causal_lm/cpp/beam_search_causal_lm.cpp

```diff
@@ -22,52 +18,187 @@ std::string detokenize(ov::InferRequest& detokenizer, const std::vector<int64_t>
     detokenizer.infer();
     return detokenizer.get_output_tensor().data<std::string>()[0];
 }
+
+std::pair<ov::Tensor, ov::Tensor> pad_left(ov::Tensor&& input_ids, ov::Tensor&& attention_mask) {
+    const size_t batch_size = input_ids.get_shape().at(0);
+    const size_t sequence_length = input_ids.get_shape().at(1);
+    int64_t* inputs_data = input_ids.data<int64_t>();
+    int64_t* attention_mask_data = attention_mask.data<int64_t>();
+
+    for (size_t batch = 0; batch < batch_size; batch++) {
+        const size_t batch_offset = batch * sequence_length;
+
+        // last token in the sequence is not a PAD_TOKEN, skipping
+        if (inputs_data[batch_offset + sequence_length - 1] != SPECIAL_TOKEN::PAD_TOKEN) {
+            continue;
+        }
+
+        size_t pad_tokens_number = 0;
+        for (int i = sequence_length - 1; i >= 0; i--) {
+            const size_t token_offset = batch_offset + i;
+
+            if (inputs_data[token_offset] == SPECIAL_TOKEN::PAD_TOKEN) {
+                continue;
+            }
+
+            if (pad_tokens_number == 0) {
+                pad_tokens_number = sequence_length - i - 1;
+            }
+
+            std::swap(inputs_data[token_offset], inputs_data[token_offset + pad_tokens_number]);
+            std::swap(attention_mask_data[token_offset], attention_mask_data[token_offset + pad_tokens_number]);
+        }
+    }
+
+    return {input_ids, attention_mask};
+}
+
+std::pair<ov::Tensor, ov::Tensor> tokenize(ov::InferRequest& tokenizer, std::vector<std::string> prompts) {
+    tokenizer.set_input_tensor(ov::Tensor{ov::element::string, {prompts.size()}, prompts.data()});
```

Comment:
side note:
currently, we use batched inference with tokenizers and it creates attention mask for us, which we need to "parse" later. Maybe, alternatively, we could tokenize prompt one by one and it will return us raw (unpadded) data, which we can use more optimally to fill position_ids, etc.

I'm not sure which solution is more optimal, so, let's stick to current one, because it's already implemented.